Adding bzlmod support document (#1214)
bazelbuild/rules_python: BZLMOD_SUPPORT.md

> # Current bzlmod support

Reviewer: ```suggestion
# Bzlmod support
```

> ## `rule_python` `bzlmod` support
> 
> - Status: Beta
> - Full Feature Parity: No
> 

Reviewer: ```suggestion
## `rule_python` `bzlmod` support

- Status: Beta
- Full Feature Parity: No

Some features are missing or broken, and the public APIs are not yet stable.
```

> ## `rule_python` `bzlmod` support
> 
> - Status: Beta
> - Full Feature Parity: No
> 

Reviewer: ```suggestion
## `rule_python` `bzlmod` support

- Status: Beta
- Full Feature Parity: No

Some features are missing or broken, and the public APIs are not yet stable.
```

> - Full Feature Parity: No
> 

Reviewer: ```suggestion
- Full Feature Parity: No

Some functionality is missing or broken and the public API is not yet stable.
```

## Configuration

The releases page will give you the latest version number, and a basic example.  The release page is located [here](/bazelbuild/rules_python/releases).

## What is bzlmod

> Bazel supports external dependencies, source files (both text and binary) used in your build that are not from your workspace. For example, they could be a ruleset hosted in a GitHub repo, a Maven artifact, or a directory on your local machine outside your current workspace.
>
> As of Bazel 6.0, there are two ways to manage external dependencies with Bazel: the traditional, repository-focused WORKSPACE system, and the newer module-focused MODULE.bazel system (codenamed Bzlmod, and enabled with the flag `--enable_bzlmod`). The two systems can be used together, but Bzlmod is replacing the WORKSPACE system in future Bazel releases.
> -- <cite>https://bazel.build/external/overview</cite>

## Examples

We have two examples that demonstrate how to configure `bzlmod`.

> The first example is located [here](examples/bzlmod), and it demonstrates the basic bzlmod configuration.

Reviewer: ```suggestion
The first example is in [examples/bzlmod](examples/bzlmod), and it demonstrates basic bzlmod configuration.
```

A user does not use `local_path_override` stanza and would define the version in the `bazel_dep` line.

> The second example demonstrates the use of `bzlmod` to configure both `rules_python` and `gazelle`.

Reviewer: ```suggestion
A second example, in [examples/bzlmod_gazelle/(link) demonstrates the use of `bzlmod` to configure `gazelle` support for `rules_python`.
```

> ## Feature Partity

Reviewer: typo: parity

This rule set does not have full feature partity with the older `WORKSPACE` type configuration:

> 1. Multiple python versions are not yet supported, as demonstrated in [this](examples/multi_python_versions) example.

Reviewer: Do we have a tracking issue? If so, lets link to it.
